### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by FintanH**
* Separate Applicative cases for `Either` into `Parallel` (i.e. Validation) and `Sequential`
* Fetching `concatMap` would take like 1s
* Fixed bug in `compile.sh` by adding `set -e`

### CODE REVIEW

File: Either/Applicative/Sequential [hunk 1]
```diff
@@ -15,7 +15,9 @@ in    λ(a : Type)
                 { Left =
                     λ(l : a) → < Left = l | Right : c >
                 , Right =
-                    let map = (./Functor a).map in λ(f : b → c) → map b c f fa
+                        let map = (./../Functor a).map
+                    
+                    in  λ(f : b → c) → map b c f fa
                 }
                 g
           }
```
Comment (sellout) [inline Either/Applicative/Sequential]: This file should be `../../Monad/extractApplicative ./Type`

File: EitherT/Applicative [hunk 2]
```diff
@@ -4,7 +4,7 @@ in  let Either = ./../Either/Type
 
 in  let EitherT = ./Type
 
-in  let EitherA = ./../Either/Applicative
+in  let EitherA = ./../Either/Applicative/Sequential
```
Comment (sellout) [inline EitherT/Applicative]: Shouldn’t we split `EitherT/Applicative` as well?

File: List/Monad [hunk 3]
```diff
@@ -1,7 +1,16 @@
     let Monad = ./../Monad/Type
 
 in  let concatMap =
-          https://ipfs.io/ipfs/QmQ8w5PLcsNz56dMvRtq54vbuPe9cNnCCUXAQp6xLc6Ccx/Prelude/List/concatMap
```
Comment (sellout) [inline List/Monad]: Gabriel changed the location, so GH URLs are preferred now – do those resolve faster?

File: Either/Monad [hunk 4]
```diff
@@ -3,7 +3,7 @@
 in  let Either = ./Type
 
 in    λ(a : Type)
-    →     ./Applicative a
+    →     ./Applicative/Sequential a
```
Comment (sellout) [inline Either/Monad]: We should have `Monad/fromPureBind` (or something) that provides default values for `map` and `ap`, then we can extract the `Applicative/Sequential` as mentioned above.

File: compile.sh [hunk 5]
```diff
@@ -1,4 +1,7 @@
 #!/bin/sh
+
+set -e
+
```
Comment (sellout) [inline compile.sh]: Oh yeah – http://redsymbol.net/articles/unofficial-bash-strict-mode/
```bash
#!/bin/bash
set -euo pipefail
IFS=$'\n\t'
```
I start all scripts that way.

File: EitherT/Applicative (hunk 2 above)
Comment (FintanH) [reply EitherT/Applicative]: Ya fair, never thought about it.

File: List/Monad (hunk 3 above)
Comment (FintanH) [reply List/Monad]: Inline: `dhall <<< "./List/Monad"  0.07s user 0.03s system 52% cpu 0.184 total`
GH: `dhall <<< "./List/Monad"  0.18s user 0.04s system 27% cpu 0.804 total`
ifps: `dhall <<< "./List/Monad"  0.16s user 0.03s system 45% cpu 0.423 total` (although I've seen it hang for a second before)

**Pull request merged**